### ISSUE TITLE
Extend to logarithmic plots

### ISSUE THREAD

**Issue by RobertRol** [JosephCrispell/addTextLabels#2]
Hi Joseph,

First of all thanks for your great R package, which I really like!

There seem to be some problems with placing the labels for logarithmic plots, though. See for instance this example

```r
library(addTextLabels)

xx=10**(-4:-1)
yy=0.5*(xx**0.5)

#this doesn't show the label
plot(xx,yy,log="xy",type="b")
addTextLabels(min(xx),min(yy),"(min,min)")

#this works
plot(log10(xx),log10(yy),type="b")
addTextLabels(log10(min(xx)),log10(min(yy)),"(min,min)")
```
A quick fix to this issue is to manually transform the variables and to perform a regular plot, like in my example above. Your function then works as expected. However, it would be great if it worked out-of-the-box for log (or log-log) plots as well.

I was quickly going over your code and saw that you use `par("usr")`. See the documentation on `par` which states

> A vector of the form c(x1, x2, y1, y2) giving the extremes of the user coordinates of the plotting region. When a logarithmic scale is in use (i.e., par("xlog") is true, see below), then the x-limits will be 10 ^ par("usr")[1:2]. Similarly for the y-axis.

So, for logarithmic plots `usr` is to be interpreted differently. I guess it is possible to fix this issue via the use of `grconvertX` and `grconvertY`.

Thanks again!
Robert

**Comment by RobertRol** [comment]
Oh, and I forgot: Your `DESCRIPTION` file uses `R (>= 3.4.4)`. Is it possible to reduce the R version number to something lower or does it really depend on R v3.4.4?

**Comment by JosephCrispell** [comment]
Hi Robert,

Thanks for pointing this issue out. Also thank you for solving it! I'll add in a check for logged axes that will adjust the text labels accordingly.

As you've noticed, the code doesn't require the latest version of R so I'll lower the requirement.

Really appreciate you using the package and flagging these issues.

Yours sincerely,

Joseph Crispell

**Comment by JosephCrispell** [comment]
Hi Robert,

I improved the addTextLabels() function to automatically detect logged axes (using the `par("xlog")` and `par("ylog")` values you suggested). It seems to work nicely with the example you provided, also works for single logged axes.

If you're interested, I found the simplest method was to check if the axes were logged, if they were I logged the appropriate input coordinates and turned off the `par()` flag. At the end of the function I return the `par()` parameters to their original values to avoid harming downstream modifications.

I've updated the `DESCRIPTION` to require `R (>= 3.0.2)`. Fairly arbitrary as the package doesn't rely upon any other libraries and uses only `base` commands, but this is the version `devtools` requires.

Let me know if you're satisfied with the changes and I'll close the issue.

Thanks again for raising this really interesting issue.

Yours sincerely,

Joseph Crispell
